src/charm.py

```diff
@@ -223,32 +313,78 @@ def _get_oathkeeper_info(self) -> Dict:
     def _promtail_error(self, event: PromtailDigestError) -> None:
         logger.error(event.message)
 
+    def _create_openfga_model(self, openfga_info: OpenfgaProviderAppData) -> None:
+        if not self.unit.is_leader():
+            logger.debug("Unit does not have leadership")
+            return
+
+        try:
+            model_id = self._admin_ui_cli.create_openfga_model(openfga_info)
+        except ExecError as err:
+            logger.error(f"Exited with code: {err.exit_code}. Stderr: {err.stderr}")
+            return
+        except Error as err:
+            logger.error(f"Something went wrong when trying to run the command: {err}")
+            return
+        except Exception as e:
+            logger.error(f"Failed to get the model id: {e}")
+            return
+
+        logger.info(f"Successfully created an openfga model: {model_id}")
+        self._set_peer_data(key=self._get_version(), data=dict(openfga_model_id=model_id))
```

Comment:
Shouldn't the rest of the units be notified of this so that they can update their config?

Reply:
What do they need to be notified for? `_handle_status_update_config` is executed by all units when the store is created, and the model id is read from `_openfga_model_id` property

Reply:
nvm, I see the case where non-leaders could update their config before the model is created - I'll push a fix. Thanks for pointing it out